torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -155,11 +156,16 @@ class ProcessGroupNCCL : public ProcessGroup {
   // The CUDA events used to sync NCCL streams
   std::unordered_map<std::string, std::vector<CUDAEvent>> ncclEvents_;
 
-  // Caches the number of GPUs available in the current system
-  int numGPUs_;
-
   // Store copy of pointer to THCState retrieved from ::at::globalContext().
   THCState* thcState_;
+
+  // ID of this process group
+  std::string processGroupId_;
+
+  // processGroupID tracking
+  std::mutex pgTrackingLock_;
```

Comment:
This should be a static mutex if you expect it to lock across process group creation.

That said, nobody should be creating process groups concurrently anyway.

Reply:
Nice catch!